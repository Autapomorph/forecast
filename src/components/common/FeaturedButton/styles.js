import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const StyledFeaturedButton = styled.span`
  color: var(--yellow);
  border: 0;

  &:hover {
    color: var(--dark-yellow);
  }
`;
