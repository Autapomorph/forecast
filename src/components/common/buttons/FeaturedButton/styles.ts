import styled from 'styled-components/macro';

import BaseIconButton from 'components/common/buttons/BaseIconButton';

export const StyledFeaturedButton = styled(BaseIconButton)`
  color: var(--yellow);

  :hover {
    color: var(--dark-yellow);
  }
`;
