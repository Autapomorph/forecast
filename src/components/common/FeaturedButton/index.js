import React, { Fragment } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import faStarSolid from '@fortawesome/fontawesome-free-solid/faStar';
import faStarRegular from '@fortawesome/fontawesome-free-regular/faStar';

import { StyledFeaturedButton } from './styles';

const FeaturedButton = ({ isFeatured = false, onAdd = () => {}, onRemove = () => {} }) => (
  <Fragment>
    {isFeatured && (
      <StyledFeaturedButton onClick={onRemove}>
        <FontAwesomeIcon icon={faStarSolid} size="lg" />
      </StyledFeaturedButton>
    )}
    {!isFeatured && (
      <StyledFeaturedButton onClick={onAdd}>
        <FontAwesomeIcon icon={faStarRegular} size="lg" />
      </StyledFeaturedButton>
    )}
  </Fragment>
);

export default FeaturedButton;
