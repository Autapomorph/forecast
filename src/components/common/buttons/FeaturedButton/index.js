import React, { Fragment } from 'react';
import { faStar as faStarSolid } from '@fortawesome/free-solid-svg-icons';
import { faStar as faStarRegular } from '@fortawesome/free-regular-svg-icons';

import { StyledFeaturedButton } from './styles';

const FeaturedButton = ({ isFeatured = false, onAdd = () => {}, onRemove = () => {} }) => (
  <Fragment>
    {isFeatured && <StyledFeaturedButton icon={faStarSolid} size="lg" onClick={onRemove} />}
    {!isFeatured && <StyledFeaturedButton icon={faStarRegular} size="lg" onClick={onAdd} />}
  </Fragment>
);

export default FeaturedButton;
