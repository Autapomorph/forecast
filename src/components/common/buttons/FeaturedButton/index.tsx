import React, { Fragment } from 'react';
import { faStar as faStarSolid } from '@fortawesome/free-solid-svg-icons';
import { faStar as faStarRegular } from '@fortawesome/free-regular-svg-icons';
import { Props as FontAwesomeIconProps } from '@fortawesome/react-fontawesome';

import { StyledFeaturedButton } from './styles';

interface IFeaturedButtonProps extends Partial<FontAwesomeIconProps> {
  isFeatured: boolean;
  onAdd?: () => void;
  onRemove?: () => void;
}

const FeaturedButton: React.FC<IFeaturedButtonProps> = ({
  isFeatured = false,
  size = 'lg',
  onAdd = () => {},
  onRemove = () => {},
  ...props
}): React.ReactElement => (
  <Fragment>
    {isFeatured && (
      // @ts-ignore
      <StyledFeaturedButton icon={faStarSolid} size={size} onClick={onRemove} {...props} />
    )}
    {!isFeatured && (
      <StyledFeaturedButton icon={faStarRegular} size={size} onClick={onAdd} {...props} />
    )}
  </Fragment>
);

export default FeaturedButton;
