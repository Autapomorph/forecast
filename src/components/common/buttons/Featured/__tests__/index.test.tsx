import React from 'react';
import { shallow, mount } from 'enzyme';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faStar as faStarSolid } from '@fortawesome/free-solid-svg-icons';
import { faStar as faStarRegular } from '@fortawesome/free-regular-svg-icons';

import FeaturedButton from '..';

describe('FeaturedButton tests', () => {
  it('renders properly', () => {
    const wrapper = shallow(<FeaturedButton isFeatured={false} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render FontAwesomeIcon', () => {
    const wrapper = mount(<FeaturedButton isFeatured={false} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(FontAwesomeIcon)).toHaveLength(1);
  });

  it('should render faStarRegular', () => {
    const wrapper = mount(<FeaturedButton isFeatured={false} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(FontAwesomeIcon).prop('icon')).toEqual(faStarRegular);
  });

  it('should render faStarSolid', () => {
    const wrapper = mount(<FeaturedButton isFeatured />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(FontAwesomeIcon).prop('icon')).toEqual(faStarSolid);
  });

  it('should handle clicks', () => {
    const onAdd = jest.fn();
    const onRemove = jest.fn();

    let wrapper = mount(<FeaturedButton isFeatured={false} onAdd={onAdd} onRemove={onRemove} />);
    wrapper.find(FontAwesomeIcon).simulate('click');
    expect(onAdd).toHaveBeenCalled();

    wrapper = mount(<FeaturedButton isFeatured onAdd={onAdd} onRemove={onRemove} />);
    wrapper.find(FontAwesomeIcon).simulate('click');
    expect(onRemove).toHaveBeenCalled();
  });
});
