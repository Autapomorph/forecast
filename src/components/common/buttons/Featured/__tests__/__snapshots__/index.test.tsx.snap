// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedButton tests renders properly 1`] = `
<Fragment>
  <styles__FeaturedButton
    icon={
      Object {
        "icon": Array [
          576,
          512,
          Array [],
          "f005",
          "M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z",
        ],
        "iconName": "star",
        "prefix": "far",
      }
    }
  />
</Fragment>
`;

exports[`FeaturedButton tests should render FontAwesomeIcon 1`] = `
.c1 {
  padding: 5px;
  border: none;
  background: none;
  color: inherit;
  cursor: pointer;
}

.c1:hover {
  color: var(--white-dark);
}

.c0 {
  color: var(--yellow);
}

.c0:hover {
  color: var(--yellow-dark);
}

<FeaturedButton
  isFeatured={false}
>
  <styles__FeaturedButton
    icon={
      Object {
        "icon": Array [
          576,
          512,
          Array [],
          "f005",
          "M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z",
        ],
        "iconName": "star",
        "prefix": "far",
      }
    }
  >
    <BaseIconButton
      className="c0"
      icon={
        Object {
          "icon": Array [
            576,
            512,
            Array [],
            "f005",
            "M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z",
          ],
          "iconName": "star",
          "prefix": "far",
        }
      }
    >
      <styles__IconButton
        className="c0"
        onClick={[Function]}
      >
        <button
          className="c1 c0"
          onClick={[Function]}
          type="button"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  576,
                  512,
                  Array [],
                  "f005",
                  "M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z",
                ],
                "iconName": "star",
                "prefix": "far",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size="lg"
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-star fa-w-18 fa-lg "
              data-icon="star"
              data-prefix="far"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </FontAwesomeIcon>
        </button>
      </styles__IconButton>
    </BaseIconButton>
  </styles__FeaturedButton>
</FeaturedButton>
`;

exports[`FeaturedButton tests should render faStarRegular 1`] = `
.c1 {
  padding: 5px;
  border: none;
  background: none;
  color: inherit;
  cursor: pointer;
}

.c1:hover {
  color: var(--white-dark);
}

.c0 {
  color: var(--yellow);
}

.c0:hover {
  color: var(--yellow-dark);
}

<FeaturedButton
  isFeatured={false}
>
  <styles__FeaturedButton
    icon={
      Object {
        "icon": Array [
          576,
          512,
          Array [],
          "f005",
          "M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z",
        ],
        "iconName": "star",
        "prefix": "far",
      }
    }
  >
    <BaseIconButton
      className="c0"
      icon={
        Object {
          "icon": Array [
            576,
            512,
            Array [],
            "f005",
            "M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z",
          ],
          "iconName": "star",
          "prefix": "far",
        }
      }
    >
      <styles__IconButton
        className="c0"
        onClick={[Function]}
      >
        <button
          className="c1 c0"
          onClick={[Function]}
          type="button"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  576,
                  512,
                  Array [],
                  "f005",
                  "M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z",
                ],
                "iconName": "star",
                "prefix": "far",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size="lg"
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-star fa-w-18 fa-lg "
              data-icon="star"
              data-prefix="far"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </FontAwesomeIcon>
        </button>
      </styles__IconButton>
    </BaseIconButton>
  </styles__FeaturedButton>
</FeaturedButton>
`;

exports[`FeaturedButton tests should render faStarSolid 1`] = `
.c1 {
  padding: 5px;
  border: none;
  background: none;
  color: inherit;
  cursor: pointer;
}

.c1:hover {
  color: var(--white-dark);
}

.c0 {
  color: var(--yellow);
}

.c0:hover {
  color: var(--yellow-dark);
}

<FeaturedButton
  isFeatured={true}
>
  <styles__FeaturedButton
    icon={
      Object {
        "icon": Array [
          576,
          512,
          Array [],
          "f005",
          "M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z",
        ],
        "iconName": "star",
        "prefix": "fas",
      }
    }
  >
    <BaseIconButton
      className="c0"
      icon={
        Object {
          "icon": Array [
            576,
            512,
            Array [],
            "f005",
            "M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z",
          ],
          "iconName": "star",
          "prefix": "fas",
        }
      }
    >
      <styles__IconButton
        className="c0"
        onClick={[Function]}
      >
        <button
          className="c1 c0"
          onClick={[Function]}
          type="button"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  576,
                  512,
                  Array [],
                  "f005",
                  "M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z",
                ],
                "iconName": "star",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size="lg"
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-star fa-w-18 fa-lg "
              data-icon="star"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </FontAwesomeIcon>
        </button>
      </styles__IconButton>
    </BaseIconButton>
  </styles__FeaturedButton>
</FeaturedButton>
`;
